src/button/ReactionToggleButton.tsx

```diff
@@ -0,0 +1,324 @@
+                autoFocus
+              />
+              <CpdButton
+                Icon={CloseIcon}
+                aria-label="close search"
+                size="sm"
+                kind="destructive"
+                onClick={() => setIsSearching(false)}
+              />
+            </Form.Root>
+            <Separator />
+          </>
+        ) : null}
+        <menu>
+          {filteredReactionSet.map((reaction) => (
+            <li className={styles.reactionPopupMenuItem} key={reaction.name}>
+              <Tooltip label={reaction.name}>
+                <CpdButton
+                  kind="secondary"
+                  className={styles.reactionButton}
+                  disabled={!canReact}
+                  onClick={() => sendReaction(reaction)}
+                >
+                  {reaction.emoji}
+                </CpdButton>
+              </Tooltip>
+            </li>
+          ))}
+        </menu>
+      </section>
+      {!isSearching ? (
+        <section>
+          <li key="search" className={styles.reactionPopupMenuItem}>
+            <Tooltip label="Search">
+              <CpdButton
+                iconOnly
+                aria-label="Open reactions search"
+                Icon={SearchIcon}
+                kind="tertiary"
+                onClick={() => setIsSearching(true)}
+              />
+            </Tooltip>
+          </li>
+        </section>
+      ) : null}
+    </div>
+  );
+}
+
+interface ReactionToggleButtonProps {
+  rtcSession: MatrixRTCSession;
+  client: MatrixClient;
+}
+
+export function ReactionToggleButton({
+  client,
+  rtcSession,
+}: ReactionToggleButtonProps): ReactNode {
+  const { raisedHands, lowerHand, reactions } = useReactions();
+  const [busy, setBusy] = useState(false);
+  const userId = client.getUserId()!;
+  const isHandRaised = !!raisedHands[userId];
+  const memberships = useMatrixRTCSessionMemberships(rtcSession);
+  const [showReactionsMenu, setShowReactionsMenu] = useState(false);
+
+  const canReact = !reactions[userId];
+
+  const sendRelation = useCallback(
+    async (reaction: ReactionOption) => {
+      const myMembership = memberships.find((m) => m.sender === userId);
+      if (!myMembership?.eventId) {
+        logger.error("Cannot find own membership event");
+        return;
+      }
+      const parentEventId = myMembership.eventId;
+      try {
+        setBusy(true);
+        await client.sendEvent(
+          rtcSession.room.roomId,
+          ElementCallReactionEventType,
+          {
+            "m.relates_to": {
+              rel_type: RelationType.Reference,
+              event_id: parentEventId,
+            },
+            emoji: reaction.emoji,
+            name: reaction.name,
+          },
+        );
+        // Do NOT close the menu after this.
+      } catch (ex) {
+        logger.error("Failed to send reaction", ex);
+      } finally {
+        setBusy(false);
+        setShowReactionsMenu(false);
+      }
```

Comment:
Looks like this `finally` block is always closing the menu ..?

Reply:
Yeah, I don't know how best to present a failure right now. Asking in standup.

Reply:
Have added a generic error box so that we can raise the error state.